video/graphics.h

```diff
@@ -716,23 +716,48 @@ void setLegacyModes(bool legacy) {
 void scrollRegion(Rect * region, uint8_t direction, int16_t movement) {
 	canvas->setScrollingRegion(region->X1, region->Y1, region->X2, region->Y2);
 	if (ttxtMode) {
-    	if (direction == 3) ttxt_instance.scroll();
-  	} else {
+		if (direction == 3) ttxt_instance.scroll();
+	} else {
+		canvas->setPenColor(tbg);
+		auto moveX = 0;
+		auto moveY = 0;
 		switch (direction) {
-			case 0:	// Right
-				canvas->scroll(movement, 0);
-		  		break;
-			case 1: // Left
-				canvas->scroll(-movement, 0);
+			case 0:		// Right
+				moveX = 1;
+				break;
+			case 1:		// Left
+				moveX = -1;
+				break;
+			case 2:		// Down
+				moveY = 1;
 				break;
-			case 2: // Down
-				canvas->scroll(0, movement);
+			case 3:		// Up
+				moveY = -1;
 				break;
-			case 3: // Up
-				canvas->scroll(0, -movement);
+			case 4:		// positive X
+				moveX = cursorBehaviour & 0x02 ? -1 : 1;
 				break;
-  		}
-  	} 
+			case 5:		// negative X
+				moveX = cursorBehaviour & 0x02 ? 1 : -1;
+				break;
+			case 6:		// positive Y
+				moveY = cursorBehaviour & 0x04 ? -1 : 1;
+				break;
+			case 7:		// negative Y
+				moveY = cursorBehaviour & 0x04 ? 1 : -1;
+				break;
+		}
+		if (moveX != 0 || moveY != 0) {
+			if (movement == 0) {
+				if (moveX != 0) {
+					movement = fontW;
+				} else {
+					movement = fontH;
+				}
+			}
```

Comment:
Acorn's version of this command differs to ours with respect to how the `movement` value is interpreted.  they only support 0 and 1, where 0 indicates movement by a single character, and 1 by a single byte (which means a variable number of pixels depending on screen mode).  values greater than 1 are treated as if they were 1

our version treats the movement value as a number of pixels.  this adds in support for `0` to indicate "whole character", which gives us slightly better compatibility with Acorn's version